### DISCUSSION TITLE
How to add a plain GeoTIFF file to the correct position in an XYZ layer?

### DISCUSSION THREAD

**Discussion by ZzWDlyy** [pelicanmapping/osgearth#2405] [Support Q&A] [answered]
When I try to add an image to an offline XYZ tile layer using GDAL, it seems to be misaligned. 
 ```
       std::string addFile = fileName.toStdString();
        osgEarth::GDALImageLayer::Options gdalOpt;
        gdalOpt.url() = addFile;
        osg::ref_ptr<osgEarth::GDALImageLayer> imgLayer = new osgEarth::GDALImageLayer(gdalOpt);
        mapNode->getMap()->addLayer(imgLayer);
```
![image](https://github.com/gwaldron/osgearth/assets/112188036/77ad32a5-b729-41d2-8c70-769fc037f07a)
However, when I open it in QGIS and other GIS software, the position appears to be fine.
![image](https://github.com/gwaldron/osgearth/assets/112188036/c3a33646-0834-4f1c-b6eb-22e8d10f45a8)
Initially, I thought it might be due to the image using the wrong coordinate system, but even after converting it from EPSG:4326 to EPSG:3857, the issue persists. The tile map I'm using is  from Google Maps. What should I do? Are there any specific parameters I need to set?
 Please help me ! 
 My environment is osgEarth 3.4 with osg 3.6.5.

**Comment by jasonbeverage** [comment]
Can you provide the file you're trying to load so we can take a look at it?

**Reply by ZzWDlyy** [reply, marked as answer]
Thank you for your reply. I have identified the issue. The different services of Google Maps may have slight variations in the alignment of their maps.
![image](https://github.com/gwaldron/osgearth/assets/112188036/57fa86b8-5e74-4167-a8e2-8e3ad068f238)
![image](https://github.com/gwaldron/osgearth/assets/112188036/f6411c44-d747-4c4d-a5a2-9549007aa74b)
